src/main/java/org/primefaces/component/datatable/DataTableRenderer.java

```diff
@@ -1342,6 +1338,15 @@ else if (column instanceof DynamicColumn) {
         writer.endElement("td");
     }
 
+    protected void encodeDefaultFieldCell(FacesContext context, DataTable table, UIColumn column, ResponseWriter writer) throws IOException {
+        Object value = UITable.createValueExprFromVarField(context, table.getVar(), column.getField()).getValue(context.getELContext());
```

Comment:
I wonder if you should wrap this up in a static `UITable` static helper that returns the value as a String run through the converter.  That way the DataExporter can use it as well.